system/Images/Handlers/ImageMagickHandler.php

```diff
@@ -408,4 +429,18 @@ protected function _text(string $text, array $options = [])
 	}
 
 	//--------------------------------------------------------------------
+	
+		//--------------------------------------------------------------------
+
+	public function _getWidth()
+	{
+		return $this->resource->getImageWidth();
```

Comment:
Can you not still use `imagesx` here, so we're not intermingling the need for an extension when the rest of the class currently doesn't need it?

Does this even work currently since the image resource is not an Imagick instance?